test/inductor/padded_tensor.py

```diff
@@ -0,0 +1,152 @@
+#
+# This file implements PaddedTensor, which is used for testing tensor subclass
+# and graph partition. This is for testing only.
+
+from typing import Any, Callable, Optional
+
+import torch
+import torch.nn.functional as F
+import torch.utils._pytree as pytree
+from torch._subclasses.fake_tensor import FakeTensorMode
+from torch.fx.experimental.proxy_tensor import disable_proxy_modes_tracing
+from torch.utils._python_dispatch import return_and_correct_aliasing
+
+
+def get_pad(shape: torch.Size, multipliers: dict[int, int]) -> tuple[int, ...]:
+    pad = [0] * (len(shape) * 2)
+    for dim, multiplier in multipliers.items():
+        if 2 * dim + 1 >= len(pad):
+            break
+        pad[2 * dim] = (shape[dim] + multiplier - 1) // multiplier * multiplier - shape[
+            dim
+        ]
+        pad[2 * dim + 1] = 0
+    return tuple(pad[::-1])
+
+
+def transform(args: Any, fn: Callable[..., Any]) -> Any:
+    flat, spec = pytree.tree_flatten(args)
+    out_flat = [fn(o) for o in flat]
+    return pytree.tree_unflatten(out_flat, spec)
+
+
+class PaddedTensor(torch.Tensor):
+    @staticmethod
+    def __new__(
+        cls,
+        tensor: torch.Tensor,
+        outer_size: Optional[torch.Size] = None,
+        outer_stride: Optional[tuple[int, ...]] = None,
+        multipliers: Optional[dict[int, int]] = None,
+        neutral_element: int = 0,
+    ) -> torch.Tensor:
+        # original size and strides before padding
+        outer_size = tensor.shape if outer_size is None else outer_size
+        outer_stride = tensor.stride() if outer_stride is None else outer_stride
+        return torch.Tensor._make_wrapper_subclass(
+            cls,
+            size=outer_size,
+            strides=outer_stride,
+            dtype=tensor.dtype,
+            device=tensor.device,
+            layout=tensor.layout,
+            requires_grad=tensor.requires_grad,
+        )
+
+    def __init__(
+        self,
+        tensor: torch.Tensor,
+        outer_size: Optional[torch.Size] = None,
+        outer_stride: Optional[tuple[int, ...]] = None,
+        multipliers: Optional[dict[int, int]] = None,
+        neutral_element: int = 0,
+    ):
+        self.multipliers = multipliers
+        self.neutral_element = neutral_element
+        self.tensor = tensor
+
+    @staticmethod
+    def from_tensor(
+        tensor: torch.Tensor,
+        multipliers: Optional[dict[int, int]] = None,
+        neutral_element: int = 0,
+    ) -> "PaddedTensor":
+        multipliers = multipliers if multipliers is not None else {}
+        padded_tensor = F.pad(
+            input=tensor,
+            pad=get_pad(tensor.shape, multipliers),
+            mode="constant",
+            value=neutral_element,
+        )
+        return PaddedTensor(
+            padded_tensor,
+            tensor.shape,
+            tensor.stride(),
+            multipliers,
+            neutral_element,
+        )
+
+    def __repr__(self) -> str:
```

Comment:
can we write a smaller unit test ? if im debugging a failure with partitioning, i would prefer to have as minimal logic as possible. if/when we commit padded tensor, there will be more padded integration tests